app/models/family.rb

```diff
@@ -229,6 +229,9 @@ def self.mappable_details
       "where people.visible = #{Family.connection.quoted_true} " \
       "and families.visible = #{Family.connection.quoted_true} " \
       "and families.deleted = #{Family.connection.quoted_false} " \
+      "and (people.share_address = TRUE " \
+      "  or  #{Person.logged_in.admin?(:view_hidden_profiles)} " \
+      "  or  people.id = #{Person.logged_in.id} )" \
```

Comment:
Actually, you don't need to explicity check `super_admin?` since it is done here: https://github.com/churchio/onebody/blob/master/app/models/concerns/person/administrator.rb#L12-L14